src/objectid.ts

```diff
@@ -298,7 +298,12 @@ export class ObjectId {
     }
 
     if (typeof id === 'string') {
-      return id.length === 12 || (id.length === 24 && checkForHexRegExp.test(id));
+      if (id.length === 12) {
+        const bytes = Buffer.from(id);
+        return bytes.byteLength === 12;
+      } else if (id.length === 24) {
+        return checkForHexRegExp.test(id);
+      }
```

Comment:
There's a chance we fall through here where we didn't before, it will still end up making it to the false at the end of the function but I think we can keep this concise:

```suggestion
      if (id.length === 12) {
        const bytes = Buffer.from(id);
        return bytes.byteLength === 12;
      }
      return id.length === 24 && checkForHexRegExp.test(id);
```

This way we fast track to false if we don't have either 12 nor a 24 length string.

Reply:
I did this earlier and it failed a BSON test case, when I explicitly checked the length first it passed

Reply:
to be specific, the "Should fail to create ObjectId due to illegal hex code" test case in `bson_test`